LFS-1094: The dialogs to create a new form should not close on clicking outside them

veronikaslc opened the pull request:
Disabled backdrop cancel propagation for accidental click outside.

Commit:
LFS-1094: The dialogs to create a new form should not close on clicking outside them

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/UserDashboard.jsx:
```diff
@@ -103,7 +103,7 @@ function UserDashboard(props) {
         }
       </Grid>
-      <Dialog fullWidth maxWidth="xs" open={open} onClose={onClose}>
+      <Dialog fullWidth maxWidth="xs" open={open} onClose={onClose} disableBackdropClick disableEscapeKeyDown>
```
I think we want to keep the escape key. From the jira task description: "The dialogs should not be click-out-to-close, **since there's also a cancel button and support for the Esc key.**"
```suggestion
      <Dialog fullWidth maxWidth="xs" open={open} onClose={onClose} disableBackdropClick>
```

veronikaslc replied:
What if you accidentally press the escape key? It's close to the number buttons on a key board.

veronikaslc committed:
LFS-1094: The dialogs to create a new form should not close on clicking outside them

Keep the escape key for dialg closing

Co-authored-by: Marta Girdea <marta.girdea@gmail.com>

sashaandjic commented:
Clicking outside the dialog is still closing it.

marta- reviewed (changes requested):
Right, this is incomplete, both the New form dialog and the Subject selector dialog need this param added.

Commit:
LFS-1094: The dialogs to create a new form should not close on clicking outside them
Applied same change to the New form dialog

sashaandjic commented:
one more
![image](https://user-images.githubusercontent.com/23243244/114900522-ef203580-9de1-11eb-97f5-07adc88dff54.png)

sashaandjic commented:
I can no longer start cards for this PR. I noticed in console the 2 new lines (highlighted) that are not showing in console of other instances:
![image](https://user-images.githubusercontent.com/23243244/114917237-b8ebb180-9df3-11eb-976b-2d0b081dc705.png)

This is how it looks in other instances ( e.g. in dev branch) that I have no problem starting:
![image](https://user-images.githubusercontent.com/23243244/114917432-efc1c780-9df3-11eb-96fc-31be471e8cfe.png)

marta- commented:
Amended wrong commit message for the 2nd commit of the PR and disabled backdrop click for the subject selector as well.

marta- committed:
LFS-1094: The dialogs to create a new form should not close on clicking outside them
Applied same change to the Subject selector dialog

sashaandjic commented:
2 more:
![image](https://user-images.githubusercontent.com/23243244/114940894-6ff62600-9e10-11eb-8b7f-1f33c19e3d4c.png)
![image](https://user-images.githubusercontent.com/23243244/114941171-cf543600-9e10-11eb-8b3a-7ff61642965c.png)

marta- committed:
LFS-1094: The dialogs to create a new form should not close on clicking outside them
Applied same change to 2 more dialog displayed by the Subject selector component

sashaandjic reviewed (approved):
tested creation of forms and change of subject. all scenarios are covered.
